target_actionnetwork/sinks.py

```diff
@@ -1,15 +1,74 @@
 """ActionNetwork target sink class, which handles writing streams."""
-from __future__ import annotations
 
 from target_actionnetwork.client import ActionNetworkSink
 
 
+class CreateAdvocacyCampaignException(Exception):
+    pass
+
+
+class CreateOutreachException(Exception):
+    pass
+
+
 class ContactsSink(ActionNetworkSink):
     """ActionNetwork target sink class for Contacts"""
 
     endpoint = "people"
     name = "Contacts"
+    advocacy_campaigns = {}
+
+    def __init__(self, target, stream_name, schema, key_properties):
+        super().__init__(target, stream_name, schema, key_properties)
+        self.get_advocacy_campaigns()
+
+    def get_advocacy_campaigns(self):
+        response = self._request("GET", "advocacy_campaigns")
+        res_json = response.json()
+        total_pages = res_json.get("total_pages")
```

Comment:
You could extract this session to another function called `_fetch_advocacy_campaign_pages `. Where you would return the `total_pages`

Reply:
it could be confusing to have a `get_advocacy_campaigns` and `_fetch_advocacy_campaign_pages`, so i prefer to not do it